plugins/reporters/spdx/src/main/kotlin/Extensions.kt

```diff
@@ -170,15 +170,16 @@ internal fun Package.toSpdxPackage(
             else -> concludedLicense.nullOrBlankToSpdxNoassertionOrNone()
         },
         licenseDeclared = declaredLicensesProcessed.toSpdxDeclaredLicense(),
-        licenseInfoFromFiles = licenseInfoResolver.resolveLicenseInfo(id)
-            .filterExcluded()
-            .filter(LicenseView.ONLY_DETECTED)
-            .map { resolvedLicense ->
-                resolvedLicense.license.takeIf { it.isValid(SpdxExpression.Strictness.ALLOW_DEPRECATED) }
-                    .nullOrBlankToSpdxNoassertionOrNone()
-            }
-            .distinct()
-            .sorted(),
+        licenseInfoFromFiles = emptyList<String>().takeIf { packageVerificationCode == null } ?:
+            licenseInfoResolver.resolveLicenseInfo(id)
+                .filterExcluded()
+                .filter(LicenseView.ONLY_DETECTED)
+                .map { resolvedLicense ->
+                    resolvedLicense.license.takeIf { it.isValid(SpdxExpression.Strictness.ALLOW_DEPRECATED) }
+                        .nullOrBlankToSpdxNoassertionOrNone()
+                }
+                .distinct()
+                .sorted(),
```

Comment:
What I don't understand about the issue is that in the cases where the `licenseInfoFromFiles` is removed in the expected results files there were obviously detected licenses, but `filesAnalyzed` is `false`. As the `packageVerificationCode` is used to set the `filesAnalyzed` property, why is there no package verification code in cases where there are detected licenses?

Reply:
In the `SpdxDocument` the reporter produces, each dependency is represented by two packages:

1. The binary package.
2. The source artifact package OR the VCS package, depending on what was scanned.

Previously, `licenseInfoFromFiles` was included redundantly for both 1 and 2. Now, it is only reported for 2. Also, analog situation for the package verification code.